### PR TITLE
VIEW FIRST: add hints and tests

### PR DESCRIPTION
**Pull request by Sirrine-Jonathan**
Approving and merging any of these pull requests will merge the changes into the set base branch.
I've set it up to be merged into the 'jons-changes' branch for now. That way they won't affect anything on 'main'.
If you want them to go into main, we can change that. We can always open a PR to merge changes in 'jons-changes' into main.

This PR specifically deals with adding tests and some hints on getting your delete function working.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -29,25 +29,26 @@ function About() {
   return <p>Home Page Content</p>;
 }
 
-class Todos extends React.Component {
+export class Todos extends React.Component {
```
Comment (Sirrine-Jonathan) [inline src/App.js:32]: exporting 'Todos' so we can import it in the test file

File: src/App.js [hunk 2]
```diff
@@ -29,25 +29,26 @@ function About() {
...
-        <form onSubmit={this.handleSubmit}>
+        <form data-testid="todo-form" onSubmit={this.handleSubmit}>
           <label htmlFor="new-todo">
           </label>
           <input
-            id="new-todo" onChange={this.handleChange} value={this.state.text} placeholder='What needs to be done?'
+            id="new-todo" data-testid="todo-input" onChange={this.handleChange} value={this.state.text} placeholder='What needs to be done?'
           />
-          <button type='#'>
+          <button type="#">
```
Comment (Sirrine-Jonathan) [inline src/App.js:51]: Changing the single quotes to double quotes is inconsequential. I removed 'type="#"' at one point and probably changed the quotes when I put it back.

File: src/App.js [hunk 3]
```diff
@@ -99,7 +117,7 @@ class TodoList extends React.Component {
     return (
       <ul>
         {this.props.items.map(item => (
-          <li key={item.id}><input type='checkbox'/>{item.text}<button onClick={this.props.delete}><img src={trashcan} alt="&#128465;"/></button></li>
+          <li key={item.id}><input type='checkbox'/>{item.text}<button onClick={(e) => this.props.delete(e, item)} aria-label="delete"><img src={trashcan} alt=""/></button></li>
```
Comment (Sirrine-Jonathan) [inline src/App.js:120]: Looks like I left in my change to pass two props to the delete function, the event and the item. That should give you another hint as to what needs to happen at the function definition of 'delete'.

File: src/App.test.js [hunk 4]
```diff
@@ -1,8 +1,78 @@
...
+    fireEvent.click(deleteButton);
+    // verify the second is gone, but the first remains
+    await waitFor(() => {
+      const firstTodo = screen.queryByText('first todo');
+      const secondTodo = screen.queryByText('second todo');
+      expect(firstTodo).toBeInTheDocument();
+      expect(secondTodo).toBeNull();
+    });
+    
+  })
```
Comment (Sirrine-Jonathan) [inline src/App.test.js:77]: These are the tests for the 'Todos' component. They will run when you run 'npm run test' in command prompt. Once you have the delete function working, these should all pass.

**Review by GemWyrm — APPROVED** [review]

**Pull request merged**